src/diffdrive_roscore.cpp

```diff
@@ -80,10 +98,24 @@ namespace roboclaw {
         motor_vel.index = 0;
         motor_vel.mot1_vel_sps = 0;
         motor_vel.mot2_vel_sps = 0;
+        motor_vel.acceleration = 0;
 
-        // Linear
-        motor_vel.mot1_vel_sps += (int) (steps_per_meter * msg.linear.x);
-        motor_vel.mot2_vel_sps += (int) (steps_per_meter * msg.linear.x);
+        // Linear acceleration
+        motor_vel.acceleration = linear_acceleration;
+
+        // Linear speed
+        double linear_speed_x = msg.linear.x;
+        if(linear_speed_x > max_linear_speed){
+            linear_speed_x = max_linear_speed;
+            ROS_WARN_STREAM_THROTTLE(15, "Linear speed clipped at max speed of " << max_linear_speed << " m/s");
+        }
+        else if (linear_speed_x < -max_linear_speed){
```

Comment:
Just to be sure, we always want abs(max) = abs(min) for the linear speed?

Reply:
The speed limits are used to reduce the energy in the system and to stop the robot faster in emergency situtation. So I used the same speed limit for both direction. Since there is a control loop to control the speed and direction of the robot I don't think we will have unstable system so for now I don't see a reason set a different speed limit of each direction.